utils/timeutil.py

```diff
@@ -7,6 +7,7 @@
 from datetime import datetime as _datetime
 import time
 import tzlocal
+import dateparser
```

Comment:
I think moving forward this is the right way to do it across all providers, may as well start now.